internal/storage/image.go

```diff
@@ -437,7 +547,6 @@ func (svc *imageService) UntagImage(systemContext *types.SystemContext, nameOrID
 			return svc.store.SetNames(img.ID, prunedNames)
 		}
 	}
-
```

Comment:
I think we should move the `removeIntermediate` function here, right before we call `DeleteImage` on the main image.

Reply:
fixed!